src/main/java/consoleapp/NotificationController.java

```diff
@@ -0,0 +1,76 @@
+package consoleapp;
+
+import datagateway.notification.NotificationManager;
+import datagateway.notification.NotificationReader;
+import datagateway.notification.SettingsManager;
+import services.notification.NotificationAdder;
+import services.notification.NotificationRemover;
+import services.notification.NotificationTracker;
+
+import java.time.Duration;
+import java.util.List;
+import java.util.Map;
+
+public class NotificationController {
+
+    private final NotificationManager notificationManager;
+    private final NotificationTracker notificationTracker;
+    private final NotificationAdder notificationAdder;
+    private final NotificationRemover notificationRemover;
+    private final SettingsManager settingsManager;
+
+    public NotificationController(NotificationManager notificationManager, NotificationTracker notificationTracker,
+                                  NotificationAdder notificationAdder,
+                                  NotificationRemover notificationRemover, SettingsManager settingsManager) {
+        this.notificationManager = notificationManager;
+        this.notificationTracker = notificationTracker;
+        this.notificationAdder = notificationAdder;
+        this.notificationRemover = notificationRemover;
+        this.settingsManager = settingsManager;
+    }
+
+    /**
+     * run notificationTracker to track the upcoming notifications
+     */
+    public void startTracking() {
```

Comment:
does this ever get run?

Reply:
For GUI, Notification Controller won't be used
It's for console app but I did not construct it.